benchmark/kmeans.py

```diff
@@ -196,7 +198,9 @@ def benchmark_data_initialization(
         # X = np.hstack([X for _ in range(5)])
         # X = np.vstack([X for _ in range(20)])
         rng = default_rng(random_state)
-        init = np.array(rng.choice(X, n_clusters, replace=False), dtype=np.float32)
+        global init
+        if init == "random":
+            init = np.array(rng.choice(X, n_clusters, replace=False), dtype=np.float32)
```

Comment:
Would it be better not to overwrite `init` and remove the `global` keyword? E.g. with potentially other changes:
```suggestion
        if init == "random":
            centroids_init = np.array(rng.choice(X, n_clusters, replace=False), dtype=np.float32)
```